### ISSUE TITLE
commands that would work without being in onyo/git

### ISSUE THREAD

**Issue by TobiasKadelka** [psyinfra/onyo#76]
Commands like `onyo cat <path>` and `onyo tree <path>` would work without the files or directories being in git/onyo, and this might be also the case for even more commands.

Should the checks in the beginning of each subcommand error out in these cases? Or, for commands that change nothing at least, should they create a warning and still work? Especially for `cat` and `tree`, these could technically work even without additional infos.

Basically, we could do:
1. hard checks, no onyo/git -> no command
2. warnings, but making commands work as much as reasonably possible for each command
3. just warn/info when this would absolutely break the command, but otherwise do for example the *normal* `cat`

**Comment by aqw** [comment]
This is a good question.

My preference is for option 1, as it's simplest: we always check for sanity. The concern I have about options 2 and 3 is that it's a slippery slope into what's "reasonable".

If the exceptions are clear cut (read-only, zero parsing) then I think the more permissive approach is also good.

**Comment by TobiasKadelka** [comment]
For `onyo cat` and `onyo tree` I will then just check if the paths/files exist, and run them.
Maybe we later add more parsing, then I might discuss again if this approach makes still sense, otherwise I like that `onyo tree .` works for example also in a folder that just contains an onyo repo, without being one.

Other commands will check then always, to prevent the possibility to accidently break the repository or messing with other files.

**Comment by TobiasKadelka** [comment]
In #86 I implement checks specifically for read-only commands.
These will force at least a valid onyo repository (having .git, .onyo), and do yaml file checks.
Other checks are ignored, like if all changes are commited and all .anchor files exist.

The PR #86 is not the final word on the topic of checks, but enables and prepares future discussions implementations of more specific checks.
